include/envoy/network/address.h

```diff
@@ -165,6 +165,39 @@ class Instance {
 
 typedef std::shared_ptr<const Instance> InstanceConstSharedPtr;
 
+/**
+ * Interface for SRV records.
+ */
+class SrvInstance {
+public:
+  virtual ~SrvInstance() {}
+
+  virtual bool operator==(const SrvInstance& rhs) const PURE;
+  bool operator!=(const SrvInstance& rhs) const { return !operator==(rhs); }
+
+  /**
+   * @return the address instance for the SRV record.
+   */
+  virtual InstanceConstSharedPtr address() const PURE;
+
+  /**
+   * @return the priority of the SRV record.
+   */
+  virtual uint32_t priority() const PURE;
```

Comment:
How are priority and weight going to be used in Envoy? I'm guessing probably not anywhere, in which case we should just not add the code for now (you can replace with TODO).